module/zfs/dsl_scan.c

```diff
@@ -3025,7 +3025,8 @@ dsl_scan_async_block_should_pause(dsl_scan_t *scn)
 	if (zfs_recover)
 		return (B_FALSE);
 
-	if (scn->scn_visited_this_txg >= zfs_async_block_max_blocks)
+	if (zfs_async_block_max_blocks != 0 && scn->scn_visited_this_txg >=
+	    zfs_async_block_max_blocks)
```

Comment:
nit: let's wrap this slightly differently to avoid splitting the `>=` comparison between lines.  How about:

```c
        if (zfs_async_block_max_blocks != 0 &&
            scn->scn_visited_this_txg >= zfs_async_block_max_blocks) {
                return (B_TRUE);
        }    
```